internal/integration/sdam_error_handling_test.go

```diff
@@ -77,14 +77,15 @@ func TestSDAMErrorHandling(t *testing.T) {
 					SetPoolMonitor(tpm.PoolMonitor).
 					// Set a 100ms socket timeout so that the saslContinue delay of 150ms causes a
 					// timeout during socket read (i.e. a timeout not caused by the InsertOne context).
-					SetSocketTimeout(100 * time.Millisecond))
+					SetConnectTimeout(100 * time.Millisecond))
```

Comment:
Optional: Consider updating the above comment to describe the new expected behavior using `SetConnectTimeout`.